frontend/cypress/support/commands.ts

```diff
@@ -70,6 +70,23 @@ declare namespace Cypress {
 }
 
 let haveCookie = Cypress.env('cookie');
+let kialiToken = Cypress.env('token');
+
+// Peserve Authorization Kiali token to set cookie before each test scenario
+const preserveKialiToken = (): void => {
+  cy.getCookie('kiali-token-aes')
+    .should('exist')
+    .then(cookie => {
+      kialiToken = cookie.value;
+      haveCookie = true;
+    });
+};
+
+const getCsrfToken = (response: Cypress.Response<any>): string => {
+  const $html = Cypress.$(response.body);
+  const csrf = $html.find('input[name=csrf]').val();
+  return csrf.toString();
+};
```

Comment:
Just curious why the change in how the cookie is gotten? i.e. going from `cy.getCookie('csrf')` to getting it from the html?

Reply:
I don't know why but `cy.getCookie('csrf')` does not work with latest version of Cypress (the method just hangs up throwing a timeout). I found and alternative way to get csrf token from official cypress examples (https://www.github.com/cypress-io/cypress-example-recipes/blob/decceecf8ff3a60e2b2f2dfa2461b4c3672949ff/examples/logging-in__csrf-tokens/cypress/e2e/logging-in-csrf-tokens-spec.cy.js#L69)